### PR TITLE
api: adding useSharedState, useStoryState

### PR DESCRIPTION
**Pull request by atanasster**
Issue:

## What I did
- isolated useSharedState in addition touseAddonState
- created new useStoryState that uses the story id as the key 
- added new useStoryState stories for examples/dev-kits

## How to test

examples/dev-kits

### CODE REVIEW

File: lib/api/src/index.tsx [hunk 1]
```diff
@@ -389,14 +389,25 @@ export function useAddonState<S>(addonId: string, defaultState?: S) {
...
 }
+
+export function useAddonState<S>(addonId: string, defaultState?: S) {
+  return useSharedState<S>(addonId, defaultState);
+}
+
+export function useStoryState<S>(defaultState?: S) {
+  const {
+    state: { storyId },
+  } = useContext(ManagerContext);
```
Comment (ndelangen) [inline lib/api/src/index.tsx]: Can use `useStorybookState` ?

**Comment by ndelangen** [comment]
WILD 👏

File: lib/api/src/index.tsx (hunk 1 above)
Comment (atanasster) [reply lib/api/src/index.tsx]: sure, just a sec

**Review by ndelangen — APPROVED** [review]

File: lib/api/src/index.tsx (hunk 1 above)
Comment (atanasster) [reply lib/api/src/index.tsx]: oops - spoken too early. There is already a useStorybookState :

https://github.com/storybookjs/storybook/blob/c35919c497ec03c8d20ec3c074a974ec04876142/lib/api/src/index.tsx#L300

**Review by ndelangen — APPROVED** [review]

**Pull request merged**